app/models/host/managed.rb

```diff
@@ -552,11 +557,34 @@ def jumpstart?
     operatingsystem.family == "Solaris" and architecture.name =~/Sparc/i rescue false
   end
 
+  def hostgroup_inherited_attributes
+    %w{puppet_proxy_id puppet_ca_proxy_id environment_id compute_profile_id}
+  end
+
+  def apply_inherited_attributes(attributes, initialized = true)
+    return nil unless attributes
+
+    new_hostgroup_id = attributes['hostgroup_id'] || attributes['hostgroup_name']
+    #hostgroup didn't change, no inheritance needs update.
+    return attributes if new_hostgroup_id.blank?
+
+    hostgroup = initialized && self.hostgroup ? self.hostgroup : Hostgroup.find(new_hostgroup_id)
+    hostgroup = Hostgroup.find(new_hostgroup_id) unless hostgroup.id == new_hostgroup_id || hostgroup.friendly_id == new_hostgroup_id
```

Comment:
Nitpick, I find these two lines rather hard to read and understand what they do with the two .find calls.  Could you have a go at tidying them up please?

One thought would be something like:

<pre>
new_hostgroup = if initialized && hostgroup && [hostgroup.id, hostgroup.friendly_id].include?(new_hostgroup_id)
                  hostgroup
                else
                  Hostgroup.find(new_hostgroup_id)
                end
</pre>

Avoiding shadowing of `hostgroup` by calling the local variable `new_hostgroup` (to match the _id variable) would also avoid needing to use `self.` and potential bugs introduced by referencing the wrong one.